app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -679,56 +645,107 @@ public void afterTextChanged(Editable s) {
         });
     }
 
-    private Callable<Cursor> getPeopleCursorGenerator() {
-        Callable<Cursor> peopleGenerator = new Callable<Cursor>() {
+    /**
+     * Refreshes recyclerView of people drawer
+     * @throws SQLException
+     */
+    public void refreshPeopleDrawer() throws SQLException {
+        // TODO Auto-generated method stub
+        peopleDrawerList.clear();
+        List<Person> people;
+        //get list of all users with recent private messages
+        //ORDER BY timestamp DESC
+        String query = "SELECT DISTINCT people.id ,people.email, people.name FROM messages " +
+                " JOIN people ON messages.recipients  LIKE ( people.id || ',%') OR messages.recipients  LIKE ('%,' || people.id)" +
+                " WHERE (LTRIM(RTRIM(messages.recipients)) LIKE '%,1' OR LTRIM(RTRIM(messages.recipients)) LIKE '1,%') AND "
+                + Person.ISBOT_FIELD + " = 0 ";
+        if (etSearchPeople.getText().toString().equals("") || etSearchPeople.getText().toString().isEmpty()) {
+            people = app.getDao(Person.class).queryBuilder()
+                    .where().eq(Person.ISBOT_FIELD, false).and()
+                    .eq(Person.ISACTIVE_FIELD, true).query();
+            //set visibility of this image false
+            ivSearchPeopleCancel.setVisibility(View.GONE);
 
-            @Override
-            public Cursor call() throws Exception {
-                // TODO Auto-generated method stub
-                List<Person> people;
-                if (etSearchPeople.getText().toString().equals("") || etSearchPeople.getText().toString().isEmpty()) {
-                    people = app.getDao(Person.class).queryBuilder()
-                            .where().eq(Person.ISBOT_FIELD, false).and()
-                            .eq(Person.ISACTIVE_FIELD, true).query();
-                    //set visibility of this image false
-                    ivSearchPeopleCancel.setVisibility(View.GONE);
+        } else {
+            people = app.getDao(Person.class).queryBuilder()
+                    .where().eq(Person.ISBOT_FIELD, false).and()
```

Comment:
since we are now doing 2 database calls for each character entered in search text, this has become really laggy. what do you think?